### ISSUE TITLE
LogContextMissingStrategy with customizable logging level

### ISSUE THREAD

**Issue by ceefour** [aws/aws-xray-sdk-java#5]
Currently `LogErrorContextMissingStrategy` uses commons-logging and always logs error:

https://github.com/aws/aws-xray-sdk-java/blob/01d52aee854baae36ad9224598ffa709290d7afa/aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/LogErrorContextMissingStrategy.java#L18

My suggestion:

1. Use `slf4j-api` instead, which is de facto standard for modern Java apps and also more performant
2. Add `LogContextMissingStrategy` which can be used to log to user-specified level. `LogErrorContextMissingStrategy` will just become a very thin subclass (for backwards compatbility) that uses `Level.ERROR`

**Comment by jamesdbowman** [comment]
Hi ceefour,

Thanks for providing an implementation of `LogContextMissingStrategy`. I'm open to including this as a feature in the SDK. Would you be willing to distill this as a pull request? It would also be great to have environment variable and system property options available to allow for runtime-bound context missing log level decisions.

As for switching to `slf4j-api`, this has been requested in the past. At this time, we are choosing to continue our use of Apache Commons Logging as it is already required and used by the AWS SDK for Java. We'd like to keep logging semantics consistent between the SDKs wherever possible.

Thanks!

**Comment by ceefour** [comment]
Yes I agree XRay v1 SDK uses Commons Logging, however v2 would use just like other AWS v2 SDK.
